src/NuGet.Core/NuGet.Packaging/Signing/Timestamp/Rfc3161TimestampProvider.cs

```diff
@@ -51,16 +51,27 @@ public Rfc3161TimestampProvider(Uri timeStampServerUrl)
         /// <summary>
         /// Timestamps data present in the TimestampRequest.
         /// </summary>
-        public Task<PrimarySignature> TimestampPrimarySignatureAsync(TimestampRequest request, ILogger logger, CancellationToken token)
+        public Task<PrimarySignature> TimestampSignatureAsync(PrimarySignature primarySignature, TimestampRequest request, ILogger logger, CancellationToken token)
         {
-            var timestampedSignature = TimestampData(request, logger, token);
-            return Task.FromResult(PrimarySignature.Load(timestampedSignature));
+            var bytes = GetEncodedTimestampCmsForData(request, logger, token);
```

Comment:
Return timestamp `SignedCms` here and pass it [here](https://github.com/NuGet/NuGet.Client/pull/2090/files#diff-0300598d24985bef7a3aaca743788176R65).